module_utils/certificate/providers/base.py

```diff
@@ -184,6 +184,9 @@ def load_from_params(cls, module, params):
         if info.get("ip"):
             info["ip"] = [ipaddress.ip_address(to_text(ip)) for ip in info["ip"] if ip]
 
+        if info.get("key_usage"):
+            info["key_usage"] = set(info["key_usage"])
```

Comment:
This doesn't preserve the order.  I think if you want to use a `set`, you should store the data as a `list`, to preserve the order, then create a temporary `set` variable copy of the list for comparison purposes only.

Reply:
For the certificates, and the providers, the order is not important. It is only important when we were comparing lists in the role, to verify if a new certificate was needed to be issue or not.

I still don't get why we need to maintain the order.

Reply:
@richm we don't need to preserve the order. We need to ensure that the order is ignored when we compare the objects. That's why @pcahyna suggested us to use a set. The order for the key_usage is actually irrelevant because it will be always stored in the same order.

Reply:
Ok, I get it.

What about the method `key_usage`?  https://github.com/linux-system-roles/certificate/blob/master/module_utils/certificate/providers/base.py#L317
should this return a `set()` instead of `[]` if no key usage was specified?

Reply:
It should. I missed that. Fixed.